src/main/java/oreo/task/Task.java

```diff
@@ -0,0 +1,165 @@
+     * @throws IllegalCommandException when command is invalid.
+     * @throws IllegalDateTimeException when date time input is invalid.
+     */
+    public static Task generateTask(String command, Scanner tokeniser) throws IllegalCommandException,
+            IllegalDateTimeException {
+        Task newTask;
+        if (!command.equals("todo") && !command.equals("deadline")
+            && !command.equals("event")) {
+            throw new IllegalCommandException("do that");
+        } else if (!tokeniser.hasNext()) {
+            throw new IllegalCommandException("process an empty task");
+        }
+        if (command.equals("todo")) {
+            String contents = tokeniser.nextLine();
+            if (contents.contains("/by")) {
+                throw new IllegalCommandException("do that for a todo,"
+                        + "are you thinking of a deadline?");
+            } else if (contents.contains("/from") || contents.contains("/to")) {
+                throw new IllegalCommandException("do that for a todo,"
+                        + "are you thinking of an event?");
+            }
+            newTask = new ToDo(contents);
+        } else if (command.equals("deadline")){
+            String contents = tokeniser.nextLine();
+            if (!contents.contains("/by")) {
+                throw new IllegalCommandException("set a deadline wihtout a \"/by\"");
+            } else if (contents.contains("/from") || contents.contains("/to")) {
+                throw new IllegalCommandException("do that for a deadline,"
+                        + "are you thinking of an event?");
+            }
+            String[] parts = contents.split(" /by ", 2);
+            String[] dateTime = TimeParser.parseInputOut(parts[1]);
+            newTask = new Deadline(parts[0], dateTime[0], dateTime[1]);
+        } else {
+            String contents = tokeniser.nextLine();
+            if (!contents.contains("/from") || !contents.contains("/to")) {
+                throw new IllegalCommandException("set an event wihtout a \"/from\" and/or \"/to\"");
+            } else if (contents.contains("/by")) {
+                throw new IllegalCommandException("do that for an event,"
+                        + "are you thinking of a deadline?");
+            }
+            String[] message = contents.split(" /from ", 2);
+            String[] fromto = message[1].split(" /to ", 2);
+            String[] fromDateTime = TimeParser.parseInputOut(fromto[0]);
+            String[] toDateTime = TimeParser.parseInputOut(fromto[1]);
+            newTask = new Event(message[0], fromDateTime[0], fromDateTime[1],
+                    toDateTime[0], toDateTime[1]);
+        }
+        return newTask;
+    }
+
+    public boolean isComplete() {
+        return this.isComplete;
+    }
+
+    /**
+     * Change mark to opposite of the current status.
+     */
+    public void switchMark() {
+        isComplete = !isComplete;
+    }
```

Comment:
Perhaps a better name here is `toggleMark`. `switchMark` made me think of switching _to_ something, but there's no parameter to specify what to switch to which confused me at first.